File: conteudos/quick_sort/QuickSort.java
package quick_sort;

public class QuickSort {
    public static void main(String[] args) {
        int[] numeros = {8, 5, 9, 1, 3 ,2, 7, 4, 6, 0};
        System.out.print("Array desordenado: ");
        printArray(numeros);
        sortedQuick(numeros, 0, numeros.length - 1);
        System.out.print("Array ordenado: ");
        printArray(numeros);
    }

    private static void printArray(int[] arr) {
        for (int i : arr) {
            System.out.print(i + " ");
        }
        System.out.println();
    }

    private static int particionar(int[] numeros, int posicaoI, int posicaoF) {
        int pivo = numeros[posicaoF];
        int i = (posicaoI - 1);
        
        for(int j=posicaoI; j<=posicaoF; j++) {
            if(numeros[j] < pivo) {
                i ++;
                int temp = numeros[i];
                numeros[i] = numeros[j];
                numeros[j] = temp;
            }
        }
        int temp = numeros[i+1];
        numeros[i+1] = numeros[posicaoF];
        numeros[posicaoF] = temp;
        return (i+1);
    }

    private static void sortedQuick(int[] numeros, int posicaoI, int posicaoF) {
        if (posicaoI < posicaoF) {
            int posicaoPivot = particionar(numeros, posicaoI, posicaoF);
            sortedQuick(numeros, posicaoI, posicaoPivot - 1);
            sortedQuick(numeros, posicaoPivot + 1, posicaoF);
        }
    }
}

File: revisao/python/quickSort.py
def quickSort(arr):
    if len(arr) < 2:
        return arr
    else:
        pivo = arr[0]
        menores = [a for a in arr if a < pivo]
        iguais = [b for b in arr if b == pivo]
        maiores = [c for c in arr if c > pivo]
        
        return quickSort(menores) + iguais + quickSort(maiores)
    
    
vetor = [1, 4, 2, 3, 6, 4, 3, 7, 8, 3, 2, 1, 6, 8, 9]
print(quickSort(vetor))
    